### PR TITLE
docs: deploy gh pages

### PR DESCRIPTION
**Pull request by tac0turtle**
## Description

- deploy docs to gh pages
- remove circle

https://cosmos.github.io/gaia/main/hub-overview/overview.html

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Documentation
+# This job builds and deploys documenation to github pages.
+# It runs on every push to master.
+on:
+  pull_request:
```
Comment (tac0turtle) [inline .github/workflows/docs.yml]:
```suggestion
```

**Review by alessio — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -134,11 +134,12 @@ distclean: clean
 build-docs:
 	@cd docs && \
 	while read p; do \
-		(git checkout $${p} && npm install && VUEPRESS_BASE="/$${p}/" npm run build) ; \
+		(git checkout $${p} && npm install && VUEPRESS_BASE="/gaia/$${p}/" npm run build) ; \
```
Comment (helder-moreira) [inline Makefile]: Not sure about adding this `gaia` prefix. Instead, we should add a file named `CNAME` to the `gh-pages` branch with the domain we will be using (`hub.cosmos.network`).

**Review by helder-moreira — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (helder-moreira) [reply Makefile]: I know it wont work under `cosmos.github.io` domain, but will under `hub.cosmos.network` which is what we want.

**Review by helder-moreira — APPROVED** [review]
LGTM

**Review by lovincyrus — COMMENTED** [review]
> * deploy docs to github pages

Thanks for working on this @marbar3778 

I noticed that we have already merged cosmos-sdk's docs github pages changes. Quick question: can we possibly use Netlify for deployment instead? It's easier for the redirects and deploy previews.

1. https://github.com/vuejs/vuepress/issues/239#issuecomment-384276251
2. https://docs.netlify.com/routing/redirects/

Deploy previews is also crucial for docs that are using vuepress-theme-cosmos. In many cases, a md syntax / codeblock error might cause the docs build to break.

Now that the new `master` is `main`, which will break all the docs URLs, we should set up a redirect rule for `/master/*`

**Comment by lovincyrus** [comment]
After some digging, I realized why @marbar3778  used gh-pages 💡:

1. https://github.com/cosmos/cosmos-sdk/tree/gh-pages
2. https://github.com/CosmWasm/docs/tree/gh-pages

We can have a better visibility of the versioned folders and ensure a persistent behavior during the version changes.

Let's make a couple of changes to the current setup.

## 1. gh-pages → netlify

- https://github.com/marketplace/actions/netlify-deploy
- select branch to deploy as `netlify-docs`

## 2. _redirects
`/master/*  /main/:splat 301!`

Ref: https://docs.netlify.com/routing/redirects/redirect-options/

With Netlify:
1. easier to manipulate redirects
2. deploy previews during dev
3. 404 handling https://github.com/cosmos/vuepress-theme-cosmos/issues/158, e.g. https://docs.cosmos.network/404

---

Netlify deployment of cosmos-sdk `gh-pages` branch: https://github.com/lovincyrus/cosmos-sdk-docs-netlify
Live: https://modest-euler-2bd497.netlify.app/

**Comment by lovincyrus** [comment]
About Netlify, is there a Cosmos Netlify account? @okwme

**Comment by tac0turtle** [comment]
> ## 1. gh-pages → netlify
> 
>     * [github.com/marketplace/actions/netlify-deploy](https://github.com/marketplace/actions/netlify-deploy)
> 
>     * select branch to deploy as `netlify-docs`

Are you suggesting this for staging or overall? We decided to go to github pages on all the repos. Keeping consistency is better than having it segregated.  

> ## 2. _redirects
> 
> `/master/* /main/:splat 301!`
> 
> Ref: [docs.netlify.com/routing/redirects/redirect-options](https://docs.netlify.com/routing/redirects/redirect-options/)
> 
> With Netlify:
> 
>     1. easier to manipulate redirects
> 
>     2. deploy previews during dev
> 
>     3. 404 handling [cosmos/vuepress-theme-cosmos#158](https://github.com/cosmos/vuepress-theme-cosmos/issues/158), e.g. [docs.cosmos.network/404](https://docs.cosmos.network/404)
> 
> 
> Netlify deployment of cosmos-sdk `gh-pages` branch: [lovincyrus/cosmos-sdk-docs-netlify](https://github.com/lovincyrus/cosmos-sdk-docs-netlify)
> Live: [modest-euler-2bd497.netlify.app](https://modest-euler-2bd497.netlify.app/)

It's already main: https://hub.cosmos.network/main/hub-overview/overview.html. Not sure when this happened, but couldn't we do the same thing we did for tendermint here? https://github.com/tendermint/tendermint/blob/master/docs/versions. 

I am opposed to go to nelify for the main deployment but could make it used for staging deployments.

**Comment by lovincyrus** [comment]
> Are you suggesting this for staging or overall?

Staging and overall. It's mainly for these reasons tbh.

- easier to manipulate redirects
- deploy previews during dev
- 404 handling cosmos/vuepress-theme-cosmos#158
   - e.g. https://docs.cosmos.network/404

> It's already main: https://hub.cosmos.network/main/hub-overview/overview.html.

Yes, we have switched to `/main`. But, we were using `/master` previously. Now that we have made the change, the old links with `/master` need to be redirected to `/main`.

> Not sure when this happened, but couldn't we do the same thing we did for tendermint here? https://github.com/tendermint/tendermint/blob/master/docs/versions.

I'm not sure this is relevant to my suggestion to update deployment from gh-pages → netlify and configure redirects for the old links (`/master`).

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Install and Build 🔧
+        run: |
+          apk add rsync
+          make build-docs LEDGER_ENABLED=false
+
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@3.7.1
```
Comment (lovincyrus) [inline .github/workflows/docs.yml:27]: What I meant was to update the action with https://github.com/marketplace/actions/netlify-deploy And change the branch name to `netlify-docs`

@marbar3778

Comment (tac0turtle) [reply .github/workflows/docs.yml:27]: I understand, but this migrates from github-pages to nelify. Cosmos-sdk and Tendermint are on github-pages, I'm not sure why gaia should be different?

I understand that something may be easier but having a unified system across all repos is better IMO.

Comment (helder-moreira) [reply .github/workflows/docs.yml:27]: I agree with @marbar3778 in having a unified system, but also understand @lovincyrus about deploy previews. Fortunately, we can have both worlds, right? We point DNS at github pages and still link Netlify to the repo for deploy previews. Anyway, I think this PR should be merged.

Comment (lovincyrus) [reply .github/workflows/docs.yml:27]: Sure, let's do that. For any new changes about Netlify, we can do that another time.

**Review by lovincyrus — APPROVED** [review]

**Review by shahankhatch — COMMENTED** [review]
Hey @marbar3778 I just realized that you were removing circle ci in this PR, but then I put in another PR for it. Do you want to remove the remove-circle-ci and liveness-test pieces, and keep only the doc building pieces for this PR?

**Pull request merged**